### DISCUSSION TITLE
Create a graphic loop, add a graphic to the stage one at a time, how to render immediately

### DISCUSSION THREAD

**Discussion by pcbren1** [pixijs/pixijs#7678] [Q&A]
Create a graphic loop, add a graphic to the stage one at a time, how to render immediately
![2021-08-06_174857](https://user-images.githubusercontent.com/86771584/128492966-b8b652fc-ab53-4bd6-a4cb-10d83907c7d5.png)

**Comment by miltoncandelero** [comment]
The problem is that your sleep is blocking the render thread. Pixi is trying to render each frame but your `while(true)` doesn't leave any time for it.

Without changing your core logic, I would make it work with some promises. 
Look at this piece of pseudocode:

```
async function draw() {
  for (...){
    ...
    await new Promise(resolve => setTimeout(resolve, 50));
  }
}
```

A promise (or even a setTimeout event) would free the main thread for a moment allowing pixi to render the screen.
